### DISCUSSION TITLE
Support for command enablement

### DISCUSSION THREAD

**Discussion by SanjulaGanepola** [eclipse-theia/theia#10457] [Q&A]
Similar to the `when` property, does Theia support the use of the `enablement` property as documented [here](https://code.visualstudio.com/api/extension-guides/command#enablement-of-commands). If so, what version of Theia is required for this?

My use case is that I have the inline action `demo.download` on all tree items in my view `demo.files`. I would like to control the enablement of this command to only allow the user to use this action on tree items with a `contextValue` of `remote`.
```
"commands": [
    {
        "command": "demo.download",
        "title": "Download",
        "enablement": "viewItem == remote"
    }
],
"menus": {
    "view/item/context": [
        {
            "command": "demo.download",
            "when": "view == demo.files",
            "group": "inline"
        }
    ]
}
```

**Comment by msujew** [comment]
Hi @SanjulaGanepola, thank you for your question.

Although I can't say for certain - as I haven't tried it so far - it seems like #8870 has added support for the `enablement` property. It is included in Theia 1.13.0.

@vince-fugnitto Do you have more info on that?

**Reply by vince-fugnitto** [reply]
> @vince-fugnitto Do you have more info on that?

I don't have additional info other that what you provided :) @SanjulaGanepola can you try the latest version and confirm if it works?

**Reply by SanjulaGanepola** [reply]
Hi @msujew @vince-fugnitto , thank you for your repsonse. I just gave this a quick try with the [tree-view-sample](https://github.com/Microsoft/vscode-extension-samples/tree/main/tree-view-sample) project by simply modifying the command contribution for `jsonOutline.renameNode` in the package.json to include the following: `"enablement": "false"`.

On VS Code, I see the expected behavior where the command is disabled where as when I tested using Eclipse Theia Blueprint (I tried both v1.13.0 and the latest v1.19.0) the command is still enabled.

**Reply by jdmchp** [reply]
@vince-fugnitto, @SanjulaGanepola   
I found that enablement is not working as I would expect for viewContainers/views.
I created this [extension](https://github.com/jdmchp/show-enablement-issue) to show the issue. Please look at the README.MD in that repo for gifs showing the difference in behavior between vscode and Theia

This is using Theia Blueprint 1.25.0. 


It also happens with:
```
$ git remote -v     
origin	https://github.com/eclipse-theia/theia (fetch)
origin	https://github.com/eclipse-theia/theia (push)
$ git rev-parse HEAD

92dcf245c498afd93e541f598e2540f8685e9287
```

**Reply by jdmchp** [reply]
This looks like a bug to me. So I created an [issue](https://github.com/eclipse-theia/theia/issues/11096)

**Reply by vince-fugnitto** [reply]
@jdmchp thanks, we were looking into the issue today and have a fix https://github.com/eclipse-theia/theia/pull/11095.

**Reply by jdmchp** [reply]
@vince-fugnitto that would be wonderful. Thanks!
